### DISCUSSION TITLE
Adding Types to Medusa Configuration Export

### DISCUSSION THREAD

**Discussion by adevinwild** [medusajs/medusa#3275] [Feature Requests]
Hi 👋
Users can customize the platform by exporting a configuration object and it's really great! 
However, there is currently no explicit documentation for the configuration object's properties, making it difficult to determine which features are accessible for customization.

I propose adding types by default to the Medusa configuration object in order to give better description for the accessible properties.

## Proposal
The addition of JSDoc types to the module is our suggestion.
Objects are exported in the medusa-config.js file. The format suggested is:

```javascript
/** @type {import('@medusajs/medusa').ConfigModule} */
```

![image](https://user-images.githubusercontent.com/68734155/219294673-b8c8143f-6571-4bfb-aee2-bfdcb07bed01.png)

> This will improve TypeScript support and allow developers to work more efficiently.

**Comment by srindom** [comment]
Think this is excellent - we should definitely do this. Is it something you'd be interested in working on?

**Reply by adevinwild** [reply]
Totally, I would love to contribute to it in order to add it 🙌🏼

**Comment by adevinwild** [comment]
**Yay! 🎉**
I've updated the `medusa-starter-default` since that's where the configuration came from according to my research (CLI , create-medusa-app)

https://github.com/medusajs/medusa-starter-default/pull/37

**Reply by olivermrbl** [reply]
Thanks for the contribution @adevinwild. I just added a comment on your PR :)

**Reply by olivermrbl** [reply]
Merged [#37](https://github.com/medusajs/medusa-starter-default/pull/37)

**Comment by srindom** [comment]
@adevinwild, thanks for contributing! If you are up for it, I think it would also be great to add JSDoc descriptions to the properties [here ](https://github.com/medusajs/medusa/blob/master/packages/medusa/src/types/global.ts#L113) so they are available as you type. Do you want to jump on this too?

**Reply by adevinwild** [reply]
With pleasure, I will take care of it during the day 👍

**Reply by adevinwild** [reply]
Hey @srindom 
I'm trying to find some info about the `database_schema` option inside the `ConfigModule`

Is it a property deprecated ? Default is put to `public` and is used a bit with some files :
https://github.com/medusajs/medusa/search?q=database_schema

Found some info about it here too : 
https://github.com/medusajs/medusa/pull/3049/commits/ad2974a746e3ddfb3a0a148964f05fef3c182cba

But it seems that this is no longer emphasized, Is it because the `search_path`parameter in a Postgres URL replaces this property?
